plugins/modules/inventory_intent.py

```diff
@@ -1559,60 +1578,62 @@ def provisioned_wired_device(self):
-                input_device_ips.remove(device_ip)
-
-        device_type = "Wired"
+        provision_wired_list = self.config[0]['provision_wired_device']
+        total_devices_to_provisioned = len(provision_wired_list)
+        device_ip_list = []
         provision_count, already_provision_count = 0, 0
 
-        if not site_name and not input_device_ips:
-            self.status = "failed"
-            self.msg = "Site/Devices are required for Provisioning of Wired Devices."
-            self.log(self.msg, "ERROR")
-            self.result['response'] = self.msg
-            return self
-
-        provision_wired_params = {
-            'siteNameHierarchy': site_name
-        }
+        for prov_dict in provision_wired_list:
+            managed_flag = False
+            device_ip = prov_dict['device_ip']
+            device_ip_list.append(device_ip)
+            site_name = prov_dict['site_name']
+            device_type = "Wired"
+            resync_retry_count = prov_dict.get("resync_retry_count", 200)
+            # This resync retry interval will be in seconds which will check device status at given interval
+            resync_retry_interval = prov_dict.get("resync_retry_interval", 2)
+
+            if not site_name and not device_ip:
+                self.status = "failed"
+                self.msg = "Site/Devices are required for Provisioning of Wired Devices."
+                self.log(self.msg, "ERROR")
+                self.result['response'] = self.msg
+                return self
 
-        for device_ip in input_device_ips:
-            try:
-                provision_wired_params['deviceManagementIpAddress'] = device_ip
-                count = 1
-                managed_flag = True
+            provision_wired_params = {
+                'deviceManagementIpAddress': device_ip,
+                'siteNameHierarchy': site_name
+            }
 
-                # Check till device comes into managed state
-                while True:
-                    response = self.get_device_response(device_ip)
-                    self.log("Device is in {0} state waiting for Managed State.".format(response['managementState']), "DEBUG")
+            # Check till device comes into managed state
+            while resync_retry_count:
+                response = self.get_device_response(device_ip)
+                self.log("Device is in {0} state waiting for Managed State.".format(response['managementState']), "DEBUG")
+
+                if (
+                    response.get('managementState') == "Managed"
+                    and response.get('collectionStatus') == "Managed"
+                    and response.get("hostname")
+                ):
+                    managed_flag = True
+                    break
+                if response.get('collectionStatus') == "Partial Collection Failure" or response.get('collectionStatus') == "Could Not Synchronize":
+                    managed_flag = False
```

Comment:
CAn we add a debug log with device ip, resync_retry_count value and retry_interval?

Reply:
added.